src/components/ListItem.jsx

```diff
@@ -5,30 +5,27 @@ import { useEffect, useState } from 'react';
 export function ListItem({ name, listToken, itemId, data }) {
 	const [checkedState, setCheckedState] = useState(data.checked);
 
-	const handleChange = (event) => {
-		// console.log(event.target.checked)
-		if (event.target.checked) {
-			setCheckedState(true);
-		} else {
-			setCheckedState(false);
-		}
-		updateItem(listToken, itemId, checkedState);
+	const handleChange = () => {
+		setCheckedState(!data.checked);
+		updateItem(listToken, itemId, !data.checked);
 	};
 
-	// useEffect(() => {
-	// 	function unCheckAfterTime() {
-	// 		const now = Date.now();
-	// 		const oneDayInSeconds = 24 * 60 * 60;
-	// 		const newDateLastPurchased = data.dateLastPurchased.seconds;
-	// 		const limit = (newDateLastPurchased + oneDayInSeconds) * 1000;
+	useEffect(() => {
+		function unCheckAfterTime() {
+			if (data.dateLastPurchased !== null) {
```

Comment:
the if conditional statement takes care of the items that have dateLastPurchased set as null (That is the value that is initialized). I tried adding an additional item to confirm our list items were working independently and without it, it broke our app). It working now.